### DISCUSSION TITLE
Passing stream output JSON to another program

### DISCUSSION THREAD

**Discussion by indeed-a-genius** [ktr0731/evans#609] [Q&A] [answered]
Hello,

I'm using evans cli on Linux to receive periodical multi-line JSONs from a gRPC server.
I'd like to pass each JSON received to another program (a PHP script ran from PHP command line).

Can evans itself redirect its output to another program?
I tried changing the "writer" value in evans config, but without success.
Also tried xargs or piping in Bash, but can't get it to work with real-time and multi-line output.

**Comment by ktr0731** [comment, marked as answer]
I don't know about your program, but how about using `jq` to pass JSON output to your program?
`-c` option modifies each JSON object on a single line.

``` sh
evans ... | jq -c . | ./your-program
```

**Reply by indeed-a-genius** [reply]
Thanks for pointing me in the right direction.
The solution was this:
`evans ... | jq -c --unbuffered . | xargs -d'\n' -L1 ./my-program`
